### PR TITLE
feat(tmux): add support for extraConfig and custom plugins

### PR DESCRIPTION
**Pull request by vdbe**
This PR adds 2 new options to the tmux module
1. extraConfig
2. plugin

## extraConfig
`programs.tmux.extraConfig` is placed after `programs.tmux.plugins` because of this there was no easy way to configure the plugin.
image

The plugin is ran on line 59 any options changed after will have no effect.

## plugin
Many people fork and customize the tmux plugin.

### CODE REVIEW

File: modules/home-manager/tmux.nix [hunk 1]
```diff
@@ -18,12 +19,33 @@ let
...
 {
   options.programs.tmux.catppuccin =
-    lib.ctp.mkCatppuccinOpt "tmux";
+    ctp.mkCatppuccinOpt "tmux"
+    // {
+      plugin = mkOption {
+        type = types.package;
+        default = plugin;
+        description = "Catppuccin tmux plugin";
+      };
```
Comment (getchoo) [inline modules/home-manager/tmux.nix:28]:
```suggestion

```

See my comment [here](https://github.com/catppuccin/nix/pull/133#discussion_r1576653324)

tl;dr: Replacing each theme source should be done in #129, not on a per module basis

File: modules/home-manager/tmux.nix [hunk 2]
```diff
@@ -18,12 +19,33 @@ let
...
   config.programs.tmux.plugins = lib.mkIf enable [
     {
-      inherit plugin;
-      extraConfig = "set -g @catppuccin_flavour '${cfg.flavour}'";
+      inherit (cfg) plugin;
+      extraConfig = concatStrings [
+        ''
+          set -g @catppuccin_flavour '${cfg.flavour}'
+        ''
+        cfg.extraConfig
```
Comment (getchoo) [inline modules/home-manager/tmux.nix:47]:
```suggestion
      extraConfig = lib.mkBefore ''
        set -g @catppuccin_flavour '${cfg.flavour}'
      '';
```
i don't think we really need to be re-exporting our parent option here either. `lib.mkBefore` can cover basically the same use case

**Review by getchoo — CHANGES_REQUESTED** [review]

File: modules/home-manager/tmux.nix (hunk 2 above)
Comment (vdbe) [reply modules/home-manager/tmux.nix:47]: The issue with `mkBefore` is that `programs.tmux.plugins` a list is. An I don't know any ~~good~~ way of modifying that `extraConfig`. Is there a way to modify it?

Comment (getchoo) [reply modules/home-manager/tmux.nix:41]: you're right, i must have had a brain fart there. this should be fine :)

**Review by getchoo — APPROVED** [review]
thanks!
